lib/utils/request.js

```diff
@@ -0,0 +1,19 @@
+module.exports = request
+
+var nets = require('nets')
+
+var Promise = require('./promise')
+
+function request (options) {
+  options.encoding = undefined
+
+  return new Promise(function (resolve, reject) {
+    nets(options, function (error, response) {
+      if (error) {
+        return reject(error)
+      }
```

Comment:
error is only set if the request fails. If there is a response, even a server error, error is null.  So we need to check for `rseponse.statusCode >= 400` here and if that’s the case, reject.

See https://github.com/hoodiehq/hoodie-client-account/blob/master/utils/request.js#L25-L30 for comparison